maven-core/src/test/java/org/apache/maven/internal/impl/PropertiesAsMapTest.java

```diff
@@ -0,0 +1,38 @@
+package org.apache.maven.internal.impl;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import org.junit.jupiter.api.Test;
+
+public class PropertiesAsMapTest
+{
+
+    @Test
+    public void testPropertiesAsMap()
+    {
+        Properties props = new Properties();
+        props.setProperty( "foo", "bar" );
+        Map<String, String> map = new HashMap<>( new PropertiesAsMap( props ) );
```

Comment:
please add assertions.

maybe also test for confirm, that PropertiesAsMap contains only has String as key and values

Reply:
Fixed